test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -560,21 +596,31 @@ func gatherInitialVMIInfo(t *testing.T, testState *kubevirtTestState) {
 		vmDisk.attachedNode, err = volumeDriver.GetNodeForVolume(vol, cmdTimeout, cmdRetry)
 		log.FailOnError(t, err, fmt.Sprintf("Failed to get node for volume %s of context %s", vol.ID, appCtx.App.Key))
 
-		vmDisk.pvcName = vmDisk.apiVol.Locator.VolumeLabels["pvc"]
-		Dash.VerifyFatal(t, vmDisk.pvcName != "", true, "PVC name found in volume labels")
+		// Don't rely on the PX volume labels to get the PVC because it could point to a template PVC in a
+		// different namespace in some abnormal cases e.g. when "clone" instead of "copy" is used to populate
+		// the volume. We avoid clones in our test specs but let's not create an unnecessary dependency.
+		pv, err := core.Instance().GetPersistentVolume(vol.ID)
+		log.FailOnError(t, err, "Failed to get PV %s of context %s", vol.ID, appCtx.App.Key)
 
-		pvc, err := core.Instance().GetPersistentVolumeClaim(vmDisk.pvcName, appCtx.App.NameSpace)
+		Dash.VerifyFatal(t, pv.Spec.ClaimRef != nil, true, fmt.Sprintf(
+			"PV %s of app %s has no claimRef", pv.Name, appCtx.App.Key))
```

Comment:
```suggestion
			"PV %s for PVC %s used by app %s has no claimRef", pv.Name, vol.ID, appCtx.App.Key))
```

Reply:
vol.ID and pv.Name are both the same. In fact, we used vol.ID to retrieve the PV a few lines above. However, your point is valid. I added apiVol.Id to the message.